packages/api/cms-api/src/dam/images/images.controller.ts

```diff
@@ -107,11 +115,15 @@ export class ImagesController {
             throw new NotFoundException();
         }
 
+        if (params.contentHash && file.contentHash !== params.contentHash) {
+            throw new Error("Content Hash mismatch!");
```

Comment:
We could throw a BadRequestException instead of an InternalServerException

Reply:
like this?
77cdc718583dbf2e914e833bd94341bcbfc5f5b7


Uploading Bad request.mov…

Reply:
should i also adjust the NotFoundException for the isValidHash to an BadRequestException?


![Screenshot 2024-10-24 at 11 05 15](https://github.com/user-attachments/assets/ea441944-e606-4fec-bb47-9c11757b4e1a)

Reply:
I'd say yes. @thomasdax98 what do you think?